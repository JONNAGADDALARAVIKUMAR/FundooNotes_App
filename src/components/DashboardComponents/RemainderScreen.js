import React, { Component } from 'react';
import {View, Text} from 'react-native';
import { Appbar } from 'react-native-paper';
import ArchivedNotesScreenStyles from '../../styles/ArchivedNotesScreenStyles';
import NotesView from '../DashboardComponents/NotesView';
import BottomBar from './BottomBar'

export default class RemainderScreen extends Component {
    constructor(props) {
        super(props)
        this.state = {
            listView: false
        }
    }

    openDrawer = () => {
        const {onPress} = this.props
        this.props.navigation.openDrawer()
        //onPress();
    }

    selectView = () => {
        const {onPress} = this.props
        this.setState({
            listView: !this.state.listView
        })
        //onPress();
    }

    render() {
        return(
            <View style = {ArchivedNotesScreenStyles.backGround_Style}>
                <Appbar style = {ArchivedNotesScreenStyles.App_Bar_Style_Bottom}>
                    <Appbar.Action
                        icon = "menu"
                        onPress = {this.openDrawer}
                    />
                    <Text style = {ArchivedNotesScreenStyles.title_Style}>
                        Remainder
                    </Text>
                    <Appbar.Content/>
                    <Appbar.Action 
                        icon = {'magnify'}
                        onPress = { () => console.log('search')}
                    />
                    <Appbar.Action 
                        icon = {(this.state.listView) ? 'view-agenda-outline' : 'view-grid-outline'}
                        onPress={this.selectView}
                    />
                </Appbar>
                <NotesView
                    navigation = {this.props.navigation} 
                    changeLayout = {this.state.listView} 
                    deletedStatus = {false}
                    remainderStatus = {true}/>

                <BottomBar 
                    navigation = {this.props.navigation}/>
            </View>
        )
    }
}