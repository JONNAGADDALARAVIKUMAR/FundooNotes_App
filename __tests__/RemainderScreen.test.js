import React from 'react';
import {shallow} from 'enzyme';
import RemainderScreen from '../src/components/DashboardComponents/RemainderScreen';

describe('test RemainderScreen Class', () => {
    it('test when render should match to snapshot', async () => {
        const component = shallow(<RemainderScreen/>)
        expect(component).toMatchSnapshot();
    })
})