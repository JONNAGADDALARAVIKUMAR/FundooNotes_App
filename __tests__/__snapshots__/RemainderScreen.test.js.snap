// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RemainderScreen Class test when render should match to snapshot 1`] = `ShallowWrapper {}`;
